src/common/transformations/src/transformations/transpose_sinking/ts_unsqueeze.cpp

```diff
@@ -181,27 +185,27 @@ TSUnsqueezeBackward::TSUnsqueezeBackward() {
         const auto& pattern_to_output = m.get_pattern_map();
 
         auto transpose = pattern_to_output.at(transpose_label);
-        auto unsqueeze = pattern_to_output.at(unsqueeze_label);
-        if (transformation_callback(unsqueeze)) {
+        auto main_node = pattern_to_output.at(unsqueeze_label);
+        if (transformation_callback(main_node)) {
             return false;
         }
 
         auto transpose_order = std::dynamic_pointer_cast<Constant>(transpose->get_input_node_shared_ptr(1));
-        auto unsqueeze_axes = std::dynamic_pointer_cast<Constant>(unsqueeze->get_input_node_shared_ptr(1));
+        auto unsqueeze_axes = std::dynamic_pointer_cast<Constant>(main_node->get_input_node_shared_ptr(1));
```

Comment:
In this transformation we expect `unsqueeze_axes` to be constant. Can we generalize it and support non-constant axes and non-constant TS order? Using Gather operation we can reorder axes for example or we need to introduce some auxiliary operation into our opset for this purpose?

Reply:
The current limitation that the axes are constant.
I created a ticket for the next release: 108225